form.html

```diff
@@ -1,53 +1,77 @@
 <!DOCTYPE html>
-<html>
-   
+<html>  
 <body>
 
 <h2>HTML Forms</h2>
 
-<form action="/action_page.php">
-  <label for="fname">First name:</label><br>
-  <input type="text" id="fname" name="fname" pattern=".{3,}"  ><br>
-  <label for="lname">Last name:</label><br>
-  <input type="text" id="lname" name="lname" pattern=".{3,}"  ><br>
-  <label for="date">Date :</label><br>
-  <input type="date" id="date" name="date"  ><br>
-  <label for="salary">Salary:</label><br>
-  <input type="number" id="salary" min="0"> <br>
-  <label for="email">email</label><br>
-  <input type="email" id="email" > <br>
-  <label for="color">Favorite Color</label><br>
-  <input type="color" id="color" > <br>
-  <label for="image">Favorite Color</label><br>
-  <input type="file" id="image" accept="image/png,image/gif,image/jpeg" ><br>
-  <p>Please select your gender</p>
+<form id="my-form" onsubmit="myFunction()">
+  <div>
+    <label for="fname">First name:</label><br>
+    <input type="text" id="fname" name="fname" pattern=".{3,}"><br>

```

Comment:
it is more readable to use minlength attribute, use pattern in case of you want to apply regex